### DISCUSSION TITLE
Sign saved file hash

### DISCUSSION THREAD

**Discussion by Stezido** [openkfw/TruBudget#859] [General]
When a file is uploaded only the docIds are saved onto the blockchain.
In terms of security this is not enough because the file located at the accesspoint can still be modified. 
To be sure that the downloaded file behind the URL is still the same a digital signature must be uploaded too. This signature can then be checked by the api if the file downloaded is the expected one.

**Comment by mayrmartin** [comment]
There is no need for signing the document hash since every singel entry on the chain is signed at its creation. Further: at the moment there is only one wallet (node-wallet) that could be used to sign the hash. Implementing the desired feature would make sense only if each user has their own wallet.
